homeassistant/components/kodi/browse_media.py

```diff
@@ -161,46 +162,60 @@ def item_payload(item, media_library):
 
     Used by async_browse_media.
     """
+    title = item["label"]
+
+    thumbnail = item.get("thumbnail")
+    if thumbnail:
+        thumbnail = media_library.thumbnail_url(thumbnail)
+
     if "songid" in item:
         media_content_type = MEDIA_TYPE_TRACK
         media_content_id = f"{item['songid']}"
+        can_play = True
+        can_expand = False
     elif "albumid" in item:
         media_content_type = MEDIA_TYPE_ALBUM
         media_content_id = f"{item['albumid']}"
+        can_play = True
+        can_expand = True
     elif "artistid" in item:
         media_content_type = MEDIA_TYPE_ARTIST
         media_content_id = f"{item['artistid']}"
+        can_play = True
+        can_expand = True
     elif "movieid" in item:
         media_content_type = MEDIA_TYPE_MOVIE
         media_content_id = f"{item['movieid']}"
+        can_play = True
+        can_expand = False
     elif "episodeid" in item:
         media_content_type = MEDIA_TYPE_EPISODE
         media_content_id = f"{item['episodeid']}"
+        can_play = True
+        can_expand = False
     elif "seasonid" in item:
         media_content_type = MEDIA_TYPE_SEASON
         media_content_id = f"{item['tvshowid']}/{item['season']}"
+        can_play = False
+        can_expand = True
     elif "tvshowid" in item:
         media_content_type = MEDIA_TYPE_TVSHOW
         media_content_id = f"{item['tvshowid']}"
+        can_play = False
+        can_expand = True
     else:
         # this case is for the top folder of each type
         # possible content types: album, artist, movie, library_music, tvshow
-        media_content_type = item.get("type")
+        media_content_type = item["type"]
         media_content_id = ""
-
-    title = item["label"]
-    can_play = media_content_type in PLAYABLE_MEDIA_TYPES and media_content_id
-    can_expand = media_content_type in EXPANDABLE_MEDIA_TYPES
-
-    thumbnail = item.get("thumbnail")
-    if thumbnail:
-        thumbnail = media_library.thumbnail_url(thumbnail)
-
-    if media_content_type == MEDIA_TYPE_MOVIE and not media_content_id:
-        media_class = MEDIA_CLASS_DIRECTORY
+        can_play = False
         can_expand = True
-    else:
+
+    try:
         media_class = CONTENT_TYPE_MEDIA_CLASS[media_content_type]
+    except KeyError as err:
+        _LOGGER.debug("Unknown media type received: %s", media_content_type)
+        raise UnknownMediaType from err
```

Comment:
We need to catch this error on line 154 and continue in the child iteration. Use a regular for loop instead of list comprehension.

Reply:
True, but couldn't I just return `None`?

Reply:
I'd need to handle that exception below in `library_payload()` as well.